cirq/sim/clifford/stabilizer_state_ch_form.py

```diff
@@ -159,27 +161,16 @@ def _Y(self, q):
         self._X(q)
         self.omega *= 1j
 
-    def _CZ(self, q, r, right=False):
```

Comment:
I'm not sure I understand why you are removing the right vs left multiplication, can you explain, I don't get the "This will allow each gate to have a definite action" argument if that's why?
As far as I see, this structure maps nicely to the paper's definition of <img src="https://render.githubusercontent.com/render/math?math=%5Cmathcal%7BL%7D%5B%5CGamma%5D"> and <!-- $\mathcal{R}[\Gamma]$ --> <img src="https://render.githubusercontent.com/render/math?math=%5Cmathcal%7BR%7D%5B%5CGamma%5D"> .

Reply:
#2948  These methods defining individual Clifford gate behavior are going to be deleted and we will rely on the implementation of `CZPowGate._act_on` to control how the `StabilizerStateCHForm` evolves. `_act_on_` does not have access to an extra right-ness argument like here and has to stick to the default "left" implementation.

So we either move the `right=True` logic to the caller (`update_sum`) right now (as this PR does) or split it out into a `_CZ_right` function which I believe is more messy.

Reply:
OK, got it. 
I have two thoughts here: 
- act_on is a "left multiplication"-only, do we want to have a right multiplication version of act_on? 
- I don't think that having these lines in themselves are readable: they are completely disconnected from the gates, and will be very hard to link them back to the original paper. At least a comment should indicate that this is the right multiplication implementation of CZ, but I think it would be better to keep it in a named private method (_CZ_right) - why do you think that it's more messy?
```
self.G[:, q] ^= self.G[:, i]
self.F[:, i] ^= self.F[:, q]
self.M[:, q] ^= self.M[:, i]
```

Reply:
Created _right methods